File: gb-angular-frontend/src/app/interfaces/refund.ts
import { OrderDetail } from './order-detail';

export interface Refund {
  refund_id: number;
  order_id: number;
  order_details_id: number;
  seller_id: number;
  reason: string;
  bank_number: string;
  bank_name: string;
  refund_amount: number;
  progress_status: string;
  requester_id: number;
  request_date: string;
  approval_date: string;
  orderDetail: OrderDetail;
  seller_username: string;
  requester_username: string;
  product_id: number;
  package_id: number;
  sell_type: string;
  grpbuy_end: string;
  refund_qty: number;
  org_amount: number;
  org_qty: number;
  unit_price: number;
  refund_type: string;

}

File: gb-spring-backend/src/main/java/com/egroupbuy/model/Refund.java
package com.egroupbuy.model;

import lombok.*;

import java.sql.Timestamp;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Refund {
    private int refund_id;
    private int order_id;
    private int order_details_id;
    private int seller_id;
    private String reason;
    private String bank_number;
    private String bank_name;
    private double refund_amount;
    private String progress_status;
    private int requester_id;
    private Timestamp request_date;
    private Timestamp approval_date;
    private OrderDetail orderDetail;
    private String seller_username;
    private String requester_username;
    private int product_id;
    private int package_id;
    private String sell_type;
    private Timestamp grpbuy_end;
    private int refund_qty;
    private double org_amount;
    private int org_qty;
    private double unit_price;
    private String refund_type;
}
